src/Uno.Foundation/AsyncOperation.TResult.cs

```diff
@@ -1,33 +1,39 @@
-﻿using Uno.Diagnostics.Eventing;
+﻿#nullable enable
+
+using Uno.Diagnostics.Eventing;
 using System;
 using System.Collections.Generic;
 using System.Text;
 using System.Threading;
 using System.Threading.Tasks;
 using System.Runtime.CompilerServices;
+using Uno;
 
 namespace Windows.Foundation
 {
-	internal class AsyncOperation<TResult> : IAsyncOperation<TResult>
+	internal class AsyncOperation<TResult> : IAsyncOperation<TResult>, IAsyncOperationInternal<TResult>
 	{
-		private CancellationTokenSource _cts = new CancellationTokenSource();
-		private AsyncOperationCompletedHandler<TResult> _onCompleted;
-		private Task<TResult> _task;
+		public static AsyncOperation<TResult> FromTask(FuncAsync<AsyncOperation<TResult>, TResult> builder)
+			=> new AsyncOperation<TResult>(builder);
+
+		private static long _nextId;
```

Comment:
Codacy found an issue: [A static field in a generic type is not shared among instances of different close constructed types.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6307767)